projects/vdk-plugins/vdk-jobs-troubleshooting/src/vdk/plugin/jobs_troubleshoot/troubleshoot_utilities/thread_dump.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import json
+import logging
+import sys
+import threading
+import traceback
+from http.server import BaseHTTPRequestHandler
+
+from vdk.internal.core.config import Configuration
+from vdk.plugin.jobs_troubleshoot.api.troubleshoot_utility import ITroubleshootUtility
+from vdk.plugin.jobs_troubleshoot.troubleshoot_configuration import (
+    TROUBLESHOOT_PORT_TO_USE,
+)
+from vdk.plugin.jobs_troubleshoot.troubleshoot_utilities.healthcheck_server import (
+    HealthCheckServer,
+)
+
+log = logging.getLogger(__name__)
+
+
+class ThreadDumpHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        if self.path == "/threads":
+            self.send_response(200)
+            self.wfile.write(b"making a thread dump")
+            log.info("Dumping threads")
+            self._log_thread_dump()
+        else:
+            self.send_error(404)
+
+    @staticmethod
+    def _log_thread_dump():
+        try:
+            for t in threading.enumerate():
+                log.info(
+                    f"Thread:{t.getName()} alive:{t.is_alive()} daemon:{t.isDaemon()}"
```

Comment:
Does the name here match the threadId in the loop below?

Reply:
yes..